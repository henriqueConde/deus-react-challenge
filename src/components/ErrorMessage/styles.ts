import styled, { DefaultTheme, css } from 'styled-components';
import { ErrorMessageProps } from './ErrorMessage';

type WrapperProps = Pick<ErrorMessageProps, 'size'>;

const wrapperModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
};

export const Wrapper = styled.p<WrapperProps>`
  ${({ theme, size }) => css`
    color: ${theme.colors.red};
    ${!!size && wrapperModifiers[size](theme)}
  `}
`;
