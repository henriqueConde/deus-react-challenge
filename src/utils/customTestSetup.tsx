/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom/server';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ThemeProvider } from 'styled-components';
import theme from '../styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const queryClient = new QueryClient();

function customRender(
  children: ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) {
  return render(
    <StaticRouter location="/">
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </QueryClientProvider>
    </StaticRouter>,
    renderOptions
  );
}

export * from '@testing-library/react';
export { customRender as render };
