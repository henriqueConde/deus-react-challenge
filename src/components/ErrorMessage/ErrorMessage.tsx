import { ReactNode } from 'react';
import * as S from './styles';

export type ErrorMessageProps = {
  children: ReactNode;
  size?: 'small' | 'large';
};

const ErrorMessage = ({ children, size = 'small' }: ErrorMessageProps) => (
  <S.Wrapper size={size}>{children}</S.Wrapper>
);

export default ErrorMessage;
