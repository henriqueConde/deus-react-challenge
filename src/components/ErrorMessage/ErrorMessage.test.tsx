import { describe, it } from 'vitest';
import { render, screen } from '@utils/customTestSetup';
import ErrorMessage from './ErrorMessage';

const mockProps = {
  children: '<p>Child element</p>',
};

describe('<ErrorMessage />', () => {
  it('should render the heading', () => {
    const { container } = render(<ErrorMessage {...mockProps} />);

    const child = screen.getByText(/Child element/i);

    expect(child).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
