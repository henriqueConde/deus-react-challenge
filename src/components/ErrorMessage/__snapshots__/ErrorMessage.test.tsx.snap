// Vitest Snapshot v1

exports[`<ErrorMessage /> > should render the heading 1`] = `
<p
  class="sc-bcXHqe jPkqUO"
>
  &lt;p&gt;Child element&lt;/p&gt;
</p>
`;
